### DISCUSSION TITLE
Topological searches in pandapower

### DISCUSSION THREAD

**Discussion by Arjunayodya** [e2nIEE/pandapower#1579] [Show and tell]
Is there any way to get the number of radial connections from each bus point using topological searches or networkx?

**Comment by SteffenMeinecke** [comment]
Hi @Arjunayodya,
yes there are topology search functionalities. Please have look to our tutorial on that: https://github.com/e2nIEE/pandapower/blob/develop/tutorials/topology.ipynb

**Comment by Arjunayodya** [comment]
yes. Actually, I already gone through thus, but I coulund find anything related to get the total number of radial connections in a network, like tze bus to which the radial is connected?

**Comment by rbolgaryn** [comment]
this is something you would need to implement yourself. You can look into the functions that determine cycles, connected components, etc. and find a way to use them to define what you require.

**Comment by marhofmann** [comment]
@Arjunayodya you could use [shortest_path](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.shortest_paths.generic.shortest_path.html#networkx.algorithms.shortest_paths.generic.shortest_path) starting from all nodes with ```degree == 1``` to the slack. the number of buses of all paths minus the number of starting nodes should be the number you are looking for
